cli/azd/pkg/exec/az_emulator.go

```diff
@@ -0,0 +1,61 @@
+package exec
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+)
+
+const (
+	emulatorEnvName string = "AZURE_AZ_EMULATOR"
+)
+
+// IsAzEmulator returns true if the AZURE_AZ_EMULATOR environment variable is defined.
+// It does not matter the value of the environment variable, as long as it is defined.
+func IsAzEmulator() bool {
+	_, emulateEnvVarDefined := os.LookupEnv(emulatorEnvName)
+	return emulateEnvVarDefined
+}
+
+// creates a copy of azd binary and renames it to az and returns the path to it
+func emulateAzFromPath() (string, error) {
+	path, err := exec.LookPath("azd")
```

Comment:
Should we be using `os.Executable()`? The `azd` in PATH is not necessarily the one being executed. The process executing may not be in PATH at all.

Reply:
That's a good idea. Not sure if it helps or not when running with debug.
Limiting to using PATH is at least a way to require azd to be installed and fail on non-standard cases.  azd installation scripts will all set azd in the PATH